awx/main/dispatch/periodic.py

```diff
@@ -0,0 +1,51 @@
+import logging
+import threading
+import time
+
+from django.conf import settings
+from django.db import connections
+from schedule import Scheduler
+
+from awx.main.dispatch.worker import TaskWorker
+
+logger = logging.getLogger('awx.main.dispatch.periodic')
+
+
+class Scheduler(Scheduler):
+
+    def run_continuously(self):
+        cease_continuous_run = threading.Event()
+        idle_seconds = max(
+            1,
+            min(self.jobs).period.total_seconds() / 2
+        )
+
+        class ScheduleThread(threading.Thread):
+            @classmethod
+            def run(cls):
+                while not cease_continuous_run.is_set():
+                    try:
+                        for conn in connections.all():
+                            # If the database connection has a hiccup, re-establish a new
+                            # connection
+                            conn.close_if_unusable_or_obsolete()
+                        self.run_pending()
+                    except Exception:
+                        logger.exception(
+                            'encountered an error while scheduling periodic tasks'
+                        )
+                    time.sleep(idle_seconds)
+
+        thread = ScheduleThread()
+        thread.daemon = True
```

Comment:
Maybe get rid of daemon attribute or else handle closing of the database connections gracefully. I'd hate for them to hang around because TCP is severed.

Reply:
It being a daemon thread just means that thread’s going to be reaped when the parent exits
With the usage of a daemon thread, I probably don’t actually need the threading event, but it gives us some extra assurances just in case this code changes or something.

What I’m trying to avoid above all else here is a scenario where the process wants to exit and it’s waiting to join on the thread.